graphql/src/main/java/com/scalar/db/graphql/datafetcher/DataFetcherBase.java

```diff
@@ -0,0 +1,185 @@
+package com.scalar.db.graphql.datafetcher;
+
+import static java.util.stream.Collectors.toList;
+
+import com.google.common.collect.ImmutableList;
+import com.scalar.db.api.ConditionalExpression;
+import com.scalar.db.api.Delete;
+import com.scalar.db.api.DistributedStorage;
+import com.scalar.db.api.DistributedTransaction;
+import com.scalar.db.api.Get;
+import com.scalar.db.api.Put;
+import com.scalar.db.api.Result;
+import com.scalar.db.api.Scan;
+import com.scalar.db.exception.storage.ExecutionException;
+import com.scalar.db.exception.transaction.TransactionException;
+import com.scalar.db.graphql.schema.Constants;
+import com.scalar.db.graphql.schema.TableGraphQlModel;
+import com.scalar.db.io.BigIntValue;
+import com.scalar.db.io.BooleanValue;
+import com.scalar.db.io.DoubleValue;
+import com.scalar.db.io.FloatValue;
+import com.scalar.db.io.IntValue;
+import com.scalar.db.io.Key;
+import com.scalar.db.io.TextValue;
+import com.scalar.db.io.Value;
+import graphql.schema.DataFetcher;
+import graphql.schema.DataFetchingEnvironment;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+abstract class DataFetcherBase<T> implements DataFetcher<T> {
+
+  protected final TableGraphQlModel tableModel;
+  protected final DistributedStorage storage;
+
+  protected DataFetcherBase(DistributedStorage storage, TableGraphQlModel tableModel) {
+    this.tableModel = tableModel;
+    this.storage = storage;
+  }
+
+  protected Key createPartitionKeyFromKeyArgument(Map<String, Object> keyArg) {
+    List<Value<?>> partitionKeyValues =
+        tableModel.getPartitionKeyNames().stream()
+            .map(name -> createValue(name, keyArg.get(name)))
+            .collect(toList());
+    return new Key(partitionKeyValues);
+  }
+
+  protected Key createClusteringKeyFromKeyArgument(Map<String, Object> keyArg) {
+    List<Value<?>> clusteringKeyValues =
+        tableModel.getClusteringKeyNames().stream()
+            .map(name -> createValue(name, keyArg.get(name)))
+            .collect(toList());
+    return new Key(clusteringKeyValues);
+  }
+
+  private Value<?> createValue(String columnName, Object inputValue) {
+    switch (tableModel.getColumnDataType(columnName)) {
+      case BOOLEAN:
+        return new BooleanValue(columnName, (Boolean) inputValue);
+      case INT:
+        return new IntValue(columnName, (Integer) inputValue);
+      case BIGINT:
+        return new BigIntValue(columnName, (Long) inputValue);
+      case FLOAT:
+        return new FloatValue(columnName, (Float) inputValue);
+      case DOUBLE:
+        return new DoubleValue(columnName, (Double) inputValue);
+      case TEXT:
+        return new TextValue(columnName, (String) inputValue);
+      case BLOB:
+      default:
+        throw new IllegalArgumentException("Invalid column");
+    }
+  }
+
+  protected Optional<Result> performGet(DataFetchingEnvironment environment, Get get)
```

Comment:
Good point. I felt like I wanted to gather them in the same place since they all do very similar things.
But now I don't think it's a very good way since they don't act as functions that should be in the base class. I moved them to each `XxxDataFetcher` class. 51155e5 Please take a look. Thank you!